schema/encoding/jsonschema.go

```diff
@@ -61,18 +48,18 @@ func ValidatorToJSONSchema(v schema.FieldValidator) string {
 			for _, value := range t.Allowed {
 				allowed = append(allowed, strconv.FormatInt(int64(value), 10))
 			}
-			result += fmt.Sprintf("%q: [ %s ],", "oneOf", strings.Join(allowed, ","))
+			result += fmt.Sprintf("%q: [ %s ],", "enum", strings.Join(allowed, ","))
 		}
-		result += fmt.Sprintf("%s,  ", BoundariesToJSONSchema(t.Boundaries))
+		result += fmt.Sprintf("%s,  ", boundariesToJSONSchema(t.Boundaries))
 	case *schema.Array:
 		result += fmt.Sprintf("%q: %q, ", "type", "array")
 		if t.ValuesValidator != nil {
-			result += fmt.Sprintf("%q: %s, ", "items", ValidatorToJSONSchema(t.ValuesValidator))
+			result += fmt.Sprintf("%q: %s, ", "items", validatorToJSONSchema(t.ValuesValidator))
 		}
 	case *schema.Object:
 		result += fmt.Sprintf("%s, ", SchemaToJSONSchema(t.Schema))
 	case *schema.Time:
-		result += fmt.Sprintf("%q: %q, ", "type", "string")
+		result += fmt.Sprintf("%q: %q, ", "type", "date-time")
```

Comment:
This is actually not quite what I meant. What I meant is that you should add date-time as a format, but keep string as the type:

```
 {
     "type": "string",
     "format": "date-time"
 }
```

See [7.  Semantic validation with "format"](http://json-schema.org/latest/json-schema-validation.html#anchor105) or more specifically [7.3.1.  date-time](http://json-schema.org/latest/json-schema-validation.html#anchor108)